### PR TITLE
fix: curio seal: Fix partial Finalize retry

### PR DESCRIPTION
**Pull request by magik6k**
## Related Issues
For now on top of https://github.com/filecoin-project/lotus/pull/11664, will rebase to master once that has landed.

## Proposed Changes
This PR fixes an edge-case in Curio finalize where somehow first try may not fully execute all steps, and on retry we'd already see some data moved. Observed on my SP setup, also confirmed the fix works.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: storage/paths/local.go [hunk 1]
```diff
@@ -501,11 +505,13 @@ func (st *Local) AcquireSector(ctx context.Context, sid storiface.SectorRef, exi
 			storiface.SetPathByType(&out, fileType, spath)
 			storiface.SetPathByType(&storageIDs, fileType, string(info.ID))
 
-			existing ^= fileType
+			existing ^= existing & fileType
```
Comment (snadrus) [inline storage/paths/local.go:508]: why xor? This is a hard to parse line.

Comment (magik6k) [reply storage/paths/local.go]: Added a helper

**Review by snadrus — APPROVED** [review]
